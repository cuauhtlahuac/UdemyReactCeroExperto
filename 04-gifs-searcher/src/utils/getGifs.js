import sortBy from 'lodash/sortBy';
export const getGifs = async category => {
	const url = `https://api.giphy.com/v1/gifs/search?q=${category ||
		'burgers'}&limit=20&api_key=${process.env.REACT_APP_GHIPY_KEY}`;
	const resp = await fetch(url);
	const { data } = await resp.json();

	const imgs = data.reduce(
		(acc, {
				id,
				title,
				url,
				images: { fixed_height },
			   }) => {
			return [
				...acc,
				{
					id,
					title,
					link: url,
					gif: fixed_height.webp,
					webp_size: Number(fixed_height.webp_size),
				},
			];
		},
		[],
	);
	
	const sortImgs = sortBy(imgs, 'webp_size');

	return sortImgs;
};
