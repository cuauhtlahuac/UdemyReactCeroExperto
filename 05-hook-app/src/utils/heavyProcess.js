const heavyProcess = (iterations) => {
  const recursion = (n) => {
    if (n === 1) return 0;
    console.log('wating for...');
    return (n = recursion(n - 1));
  };

  recursion(iterations);
  return `${iterations} iteraciones realizadas`;
};

export default heavyProcess;
