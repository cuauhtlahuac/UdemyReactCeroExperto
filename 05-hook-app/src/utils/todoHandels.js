export const handleSubmit = (e, dispatch, reset, description) => {
	e.preventDefault();

	if (description.trim().length <= 1) return;

	const newTodo = {
		id: new Date().getTime(),
		task: description,
		done: false,
	};

	const addTodo = { type: 'addTodo', payload: newTodo };

	dispatch(addTodo);
	reset();
};

export const handleDelete = (id, dispatch) => {
	const removeTodo = { type: 'removeTodo', payload: id };

	dispatch(removeTodo);
};

export const handleComplete = (id, dispatch) => {
	const completeTodo = { type: 'completeTodo', payload: id };

	dispatch(completeTodo);
};
