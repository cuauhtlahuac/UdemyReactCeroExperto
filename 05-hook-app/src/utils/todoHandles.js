import { todoActions } from "./constants/todoActions";

const { ADD_TODO, COMPLETE_TODO, REMOVE_TODO} = todoActions

export const handleSubmit = (e, dispatch, reset, description) => {
	e.preventDefault();

	if (description.trim().length <= 1) return;

	const newTodo = {
		id: new Date().getTime(),
		task: description,
		done: false,
	};

	const addTodo = { type: ADD_TODO, payload: newTodo };

	dispatch(addTodo);
	reset();
};

export const handleDelete = (id, dispatch) => {
	const removeTodo = { type: REMOVE_TODO, payload: id };

	dispatch(removeTodo);
};

export const handleComplete = (id, dispatch) => {
	const completeTodo = { type: COMPLETE_TODO, payload: id };

	dispatch(completeTodo);
};
