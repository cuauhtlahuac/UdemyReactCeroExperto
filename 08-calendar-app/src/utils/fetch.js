const baseURL = process.env.REACT_APP_API_URL;

export const noTokenFetch = async (endPoint, data, method = 'GET') => {
	const url = `${baseURL}/${endPoint}`;

	let response;

	if (method !== 'GET') {
		response = await fetch(url, {
			method,
			headers: {
				'Content-Type': 'application/json',
			},
			body: JSON.stringify(data),
		});
	} else {
		response = await fetch(url);
	}

	return await response.json();
};

export const tokenFetch = async (endPoint, data, method = 'GET') => {
	const token = localStorage.getItem('token');

	const url = `${baseURL}/${endPoint}`;

	let response;

	if (method !== 'GET') {
		response = await fetch(url, {
			method,
			headers: {
				'Content-Type': 'application/json',
				"x-token": token,
			},
			body: JSON.stringify(data),
		});
	} else {
		response = await fetch(url, {
			method,
			headers: {
				"x-token": token,
			},
		});
	}

	return await response.json();
};
