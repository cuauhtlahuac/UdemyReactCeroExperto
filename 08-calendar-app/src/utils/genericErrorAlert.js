import Swal from 'sweetalert2';

export function genericError() {
	return Swal.fire(
		'Something went wrong',
		'Please try again in few minutes.<br/><br/> If the problem persist please contact to admin',
		'error',
	);
}