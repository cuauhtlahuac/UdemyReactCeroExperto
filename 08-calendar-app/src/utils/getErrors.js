export const getErrorsMsgs = body => {
	let errorMsgs = body.msg || '';

  const { errors } = body;

	if (errors) {
		for (let error in errors) {
			errorMsgs += `${errors[error].msg}<br/><br/>`;
		}
	}

	return errorMsgs;
};
