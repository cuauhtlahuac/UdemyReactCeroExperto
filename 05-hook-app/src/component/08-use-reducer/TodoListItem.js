import React from 'react';
import PropTypes from 'prop-types';
import { FaTrashAlt, FaCheck } from 'react-icons/fa';

import { handleComplete, handleDelete } from 'utils/todoHandels';

const TodoListItem = ({ todo, index, dispatch }) => (
	<li key={todo.id} className="list-group-item">
		<p className={`${todo.done && 'font-weight-lighter complete'}`}>
			{`${index + 1} .- ${todo.task}`}
		</p>

		<span>
			<button
				onClick={() => handleComplete(todo.id, dispatch)}
				className={`btn ${todo.done
					? 'btn-success'
					: 'btn-outline-secondary'} mr-2`}
			>
				<FaCheck color="white" />
			</button>

			<button className="btn btn-danger">
				<FaTrashAlt
					color="#fdd"
					onClick={() => handleDelete(todo.id, dispatch)}
				/>
			</button>
		</span>
	</li>
);

TodoListItem.propTypes = {
	todo: PropTypes.object.isRequired,
	index: PropTypes.number,
	dispatch: PropTypes.func.isRequired,
};

export default TodoListItem;
