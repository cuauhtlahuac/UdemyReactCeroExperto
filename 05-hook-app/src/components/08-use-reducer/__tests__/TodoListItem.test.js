import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { mockTodo } from 'fixtures/todoMock';
import { handleComplete, handleDelete } from 'utils/todoHandles';
import TodoListItem from '../TodoListItem';

describe('TodoListItem Tests', () => {
	const props = { todo: mockTodo[0], index: 0, dispatch: jest.fn() };

	const component = mount(<TodoListItem {...props} />);

	afterAll(() => {
		component.unmount();
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});

	it('should call handleComplete function', () => {
		const button = component.find('.btn-outline-secondary');
		const handleCompleteMock = jest.fn(handleComplete);

		button.props().onClick(handleCompleteMock(1, props.dispatch));

		expect(handleCompleteMock).toBeCalled();
		expect(handleCompleteMock).toBeCalledWith(1, props.dispatch);
	});

	it('should show text correctly', () => {
		const paragraph = component.find('p');

		expect(paragraph.first().text()).toBe(
			`${props.index + 1} .- ${props.todo.task}`,
		);
	});

	it('should show "complete" className', () => {
		const newProps = {
			...props,
			todo: { ...props.todo, done: true, task: 'learn everything' },
		};

		component.setProps({ ...newProps });

		expect(component).toMatchSnapshot();

		const paragraph = component.find('p');
		expect(paragraph.hasClass('complete')).toBe(true); // hasClass refiere al className
	});

	it('should call handleDelete function', () => {
		const button = component.find('.btn-danger');
		const handleDeleteMock = jest.fn(handleDelete);

		button.props().onClick(handleDeleteMock(1, props.dispatch));

		expect(handleDeleteMock).toBeCalled();
		expect(handleDeleteMock).toBeCalledWith(1, props.dispatch);
	});
});
