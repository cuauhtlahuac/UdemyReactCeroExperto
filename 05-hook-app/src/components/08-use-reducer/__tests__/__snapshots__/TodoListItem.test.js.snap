// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem Tests should render correctly 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
  >
    1 .- Aprender React
  </p>
  <span>
    <button
      className="btn btn-outline-secondary mr-2"
      onClick={[Function]}
    >
      <svg
        color="white"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          key="0"
        />
      </svg>
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      <svg
        color="#fdd"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#fdd",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          key="0"
        />
      </svg>
    </button>
  </span>
</li>
`;

exports[`TodoListItem Tests should show "complete" className 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="font-weight-lighter complete"
  >
    1 .- learn everything
  </p>
  <span>
    <button
      className="btn btn-success mr-2"
      onClick={[Function]}
    >
      <svg
        color="white"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          key="0"
        />
      </svg>
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      <svg
        color="#fdd"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#fdd",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          key="0"
        />
      </svg>
    </button>
  </span>
</li>
`;
